### PR TITLE
cd: create new workflows #3

### PR DESCRIPTION
**Pull request by SantiagoGaonaC**

### CODE REVIEW

**Review by shoriwe — COMMENTED** [review]
Great job @SantiagoGaonaC, the pipelines seen overall correct, could you please create a new one that targets `PR`s to `dev` branch, runs all the tests as a filter to prevent broken code in upcoming releases

Create something similar to [testing.yaml](https://github.com/hawks-atlanta/gateway-java/blob/main/.github/workflows/testing.yml). But with your testing command similar to the one you put in `coverage.yaml`

```shell
dotnet test --no-restore --verbosity normal /p:CollectCoverage=true
```

File: .github/workflows/coverage.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+on:
+  push:
+    branches: ["main"]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
```
Comment (shoriwe) [inline .github/workflows/coverage.yaml:11]: Looking at some references of `dotnet` dependency we may need add this step in the `build` job

https://github.com/actions/setup-dotnet

File: .github/workflows/testing.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: 6.0
+
+      - name: Restore dependencies
+        run: dotnet restore
+
+      - name: Build and Test
+        run: dotnet test --no-restore --verbosity normal /p:CollectCoverage=true
```
Comment (shoriwe) [inline .github/workflows/testing.yaml]: We may not need to collect coverage during tests, removing the coverage recollection in this step should make the pipe execution faster

**Comment by SantiagoGaonaC** [comment]
> Great job @SantiagoGaonaC, the pipelines seen overall correct, could you please create a new one that targets `PR`s to `dev` branch, runs all the tests as a filter to prevent broken code in upcoming releases
> 
> Create something similar to [testing.yaml](https://github.com/hawks-atlanta/gateway-java/blob/main/.github/workflows/testing.yml). But with your testing command similar to the one you put in `coverage.yaml`
> 
> ```shell
> dotnet test --no-restore --verbosity normal /p:CollectCoverage=true
> ```

Resolved ✅

**Review by shoriwe — APPROVED** [review]
Neat solution. ready for `dev`

**Pull request merged**